packages/inter-protocol/test/supports.js

```diff
@@ -64,10 +64,20 @@ export const setUpZoeForTest = (setJig = () => {}) => {
 };
 harden(setUpZoeForTest);
 
-export const mockChainStorageRoot = () => {
-  const toStorage = v => v;
-  return makeChainStorageRoot(toStorage, 'swingset', 'mockChainStorageRoot');
+export const makeMockChainStorageRoot = () => {
+  const { rootNode, data } = makeFakeStorageKit('mockChainStorageRoot');
+  return Far('mockChainStorage', {
+    ...rootNode,
+    getBody: path => {
+      const dataStr = data.get(path);
+      assert(dataStr, `no data at ${path}`);
+      assert.typeof(dataStr, 'string');
+      const datum = JSON.parse(dataStr);
+      return JSON.parse(datum.body);
+    },
+  });
```

Comment:
This function currently relies on its callers to make assumptions about capdata, but I think it would be better to handle more of that for them. For example:
```suggestion
  const { rootNode, data } = makeTransientStorageKit('mockChainStorageRoot');
  
  // Default to a deserialize pass-by-presence objects into { iface } representations.
  // Note that this is **not** a null transformation; capdata `@qclass` and `index` properties
  // are dropped and `iface` is _added_ for repeat references.
  const defaultMarshaller = makeMarshal(undefined, (_slotId, iface) => ({ iface }));
  
  return Far('mockChainStorage', {
    ...rootNode,
    getBody: (path, marshaller = defaultMarshaller) => {
      const dataStr = data.get(path);
      assert(dataStr, `no data at ${path}`);
      assert.typeof(dataStr, 'string');
      const datum = JSON.parse(dataStr);
      return marshaller.unserialize(datum);
    },
  });
```

```patch
--- packages/inter-protocol/test/amm/vpool-xyk-amm/test-xyk-amm-swap.js
+++ packages/inter-protocol/test/amm/vpool-xyk-amm/test-xyk-amm-swap.js
@@ -1242,14 +1242,14 @@
   t.like(mockChainStorage.getBody('mockChainStorageRoot.amm.pool0.metrics'), {
     centralAmount: {
       brand: { iface: 'Alleged: central brand' },
-      value: { digits: '50000' },
+      value: 50000n,
     },
     liquidityTokens: {
       brand: { iface: 'Alleged: MoolaLiquidity brand' },
-      value: { digits: '50000' },
+      value: 50000n,
     },
     secondaryAmount: {
       brand: { iface: 'Alleged: moola brand' },
-      value: { digits: '10000' },
+      value: 10000n,
     },
   });
```

Reply:
NIce!